gtfs_realtime_translators/translators/path_rail.py

```diff
@@ -109,33 +149,67 @@ def __call__(self, data):
         return FeedMessage.create(entities=entities)
 
     @classmethod
-    def __make_trip_updates(cls, data):
-        trip_updates = []
-        now = int(pendulum.now().timestamp())
+    def __to_unix_time(cls, time):
+        datetime = int(pendulum.from_format(time, 'HH:mm').in_tz(cls.TIMEZONE).timestamp())
+        return datetime
 
-        arrivals = data['results']
-        for arrival in arrivals:
-            arrival_updates = arrival['messages']
-            for idx, update in enumerate(arrival_updates):
-                try:
-                    route_id = cls.ROUTE_ID_LOOKUP[update['lineColor']]
-                    destination = arrival['target']
-                    station = arrival['consideredStation']
+    @classmethod
+    def __route_lookup(cls,service_id, station_shortkey, track_id, destination):
+        service_mapping = cls.SERVICE_LOOKUP.get(str(service_id))
+        station_mapping = cls.STOP_ID_LOOKUP.get(station_shortkey)
+        route_id = service_mapping.get('route_id')
+        headsign = service_mapping.get(destination)
+        stop_id = station_mapping.get("tracks").get(track_id)
+        stop_name = station_mapping.get("stop_name")
+        return {
+           'route_id': route_id,
+            'stop_id': stop_id,
+            'headsign': headsign,
+            'stop_name': stop_name
+        }
 
-                    key = route_id + '_' + destination + '_' + station
-                    stop_id = cls.STOP_ID_LOOKUP[key]
-                    arrival_time = now + math.floor(update['secondsToArrival'] / 60) * 60
-                    headsign = update['headSign']
+    @classmethod 
+    def __should_skip_update(cls,route_data):
+        should_skip = False
+        for key, value in route_data.items():
+            if value is None:
+                should_skip = True
+                break
+        return should_skip    
 
+    @classmethod
+    def __make_trip_updates(cls, data):
+        trip_updates = []
+        stations = data['stations']
+        for station in stations:
+            station_shortkey = station['abbrv']
+            tracks = station.get('tracks')
+            if station_shortkey == 'systemwide':
+                continue
+            for track in tracks:
+                track_id = track.get('trackId')
+                trains = track.get('trains')
```

Comment:
Can have as below:
trains = track.get('trains', {})

Reason is: trains = track.get('trains') gives trains as None (unless, it is initialized while creating trains in track) and that will mostly result in exception at next line while using it as `enumerate(trains)`